be/src/storage/rowset/segment_v2/storage_page_data_decoder.h

```diff
@@ -0,0 +1,115 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#pragma once
+
+#include <memory>
+
+#include "common/status.h"
+#include "gen_cpp/segment_v2.pb.h"
+#include "storage/rowset/segment_v2/binary_dict_page.h"
+#include "storage/rowset/segment_v2/bitshuffle_page.h"
+#include "storage/rowset/segment_v2/page_handle.h"
+#include "storage/rowset/segment_v2/page_pointer.h"
+#include "util/slice.h"
+
+namespace starrocks::segment_v2 {
+
+class DataDecoder {
+public:
+    DataDecoder() = default;
+    virtual ~DataDecoder() = default;
+
+    virtual void reserve_head(uint8_t head_size) {}
+
+    virtual Status decode_data_page(PageFooterPB* footer, uint32_t footer_size, EncodingTypePB encoding,
```

Comment:
I think decode_page_data is more reasonable?

Reply:
done